doc/usage/restructuredtext/directives.rst

```diff
@@ -150,7 +150,8 @@ tables of contents.  The ``toctree`` directive is the central element.
 
    The special entry name ``self`` stands for the document containing the
    toctree directive.  This is useful if you want to generate a "sitemap" from
-   the toctree.
+   the toctree. The special document names ``genindex``, ``modindex``, and
+   ``search`` can also be used.
```

Comment:
Perhaps expand this, as the special names can also be used indirectly, as you illustrate in the test.